api/v1alpha1/integrationtestscenario_conversion.go

```diff
@@ -0,0 +1,95 @@
+/*
+Copyright 2023.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"reflect"
+
+	"github.com/redhat-appstudio/integration-service/api/v1beta1"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/conversion"
+)
+
+func (r *IntegrationTestScenario) SetupWebhookWithManager(mgr ctrl.Manager) error {
+	return ctrl.NewWebhookManagedBy(mgr).
+		For(r).
+		Complete()
+}
+
+// Hub marks this type as a conversion hub.
+// ConvertTo converts this ITS to the Hub version (v1beta1).
+func (src *IntegrationTestScenario) ConvertTo(dstRaw conversion.Hub) error {
+	dst := dstRaw.(*v1beta1.IntegrationTestScenario)
+	dst.ObjectMeta = src.ObjectMeta
+	dst.Spec.Application = src.Spec.Application
+	if !reflect.ValueOf(src.Spec.Environment).IsZero() {
+		dst.Spec.Environment = v1beta1.TestEnvironment(src.Spec.Environment)
+	}
+	if src.Spec.Params != nil {
+		for _, par := range src.Spec.Params {
+			dst.Spec.Params = append(dst.Spec.Params, v1beta1.PipelineParameter(par))
+		}
+	}
+	if src.Spec.Contexts != nil {
+		for _, par := range src.Spec.Contexts {
+			dst.Spec.Contexts = append(dst.Spec.Contexts, v1beta1.TestContext(par))
+		}
+	}
+	dst.Spec.ResolverRef = v1beta1.ResolverRef{
+		Resolver: "bundles",
+		Params: []v1beta1.ResolverParameter{
+			{
+				Name:  "bundle",
+				Value: src.Spec.Bundle,
+			},
+			{
+				Name:  "name",
+				Value: src.Spec.Pipeline,
+			},
+			{
+				Name:  "kind",
+				Value: "pipeline",
+			},
+		},
+	}
+	return nil
+}
+
+func (dst *IntegrationTestScenario) ConvertFrom(srcRaw conversion.Hub) error {
+	src := srcRaw.(*v1beta1.IntegrationTestScenario)
+	dst.ObjectMeta = src.ObjectMeta
+	dst.Spec.Application = src.Spec.Application
+	if !reflect.ValueOf(dst.Spec.Environment).IsZero() {
+		dst.Spec.Environment = TestEnvironment(src.Spec.Environment)
+	}
+	if dst.Spec.Params != nil {
+		for _, par := range src.Spec.Params {
+			dst.Spec.Params = append(dst.Spec.Params, PipelineParameter(par))
+		}
+	}
+	if dst.Spec.Contexts != nil {
+		for _, par := range src.Spec.Contexts {
+			dst.Spec.Contexts = append(dst.Spec.Contexts, TestContext(par))
+		}
+	}
+
+	dst.Spec.Pipeline = src.Spec.ResolverRef.Params[1].Value
```

Comment:
We do these two line only when `src.Spec.ResolverRef.Resolver == 'bundles'`

Reply:
There should not be any other type of resolver present from previous version.

Reply:
Do you have any specific way in mind to allign those attributes?

Reply:
Here `dst` is CR in `v1alpha1` version, `src` is CR in `v1beta1`, This function is to convert CR from `v1beta1` to `v1alpha1` version.

And I just realize we can't use `src.Spec.ResolverRef.Params[1].Value` because the `Params` is a list https://tekton.dev/docs/pipelines/bundle-resolver/#task-resolution so we can't make sure the first key-value pair is for `bundle`. So I think we need to find the their value by its key.

Reply:
Agreed, I think we still need to address this by looping through `src.Spec.ResolverRef.Params` and checking the `name` of the `param` before setting it as either `bundle` or `pipeline`.

Reply:
Updated.

Reply:
Thanks, I believe that this resolves this conversation